File: src/main/java/BooleanOperatorAction.java
import java.util.Objects;

import com.intellij.codeInsight.intention.PsiElementBaseIntentionAction;
import com.intellij.codeInspection.util.IntentionFamilyName;
import com.intellij.codeInspection.util.IntentionName;
import com.intellij.openapi.editor.Editor;
import com.intellij.openapi.project.Project;
import com.intellij.psi.JavaPsiFacade;
import com.intellij.psi.JavaTokenType;
import com.intellij.psi.PsiBinaryExpression;
import com.intellij.psi.PsiElement;
import com.intellij.psi.PsiElementFactory;
import com.intellij.psi.PsiMethodCallExpression;
import com.intellij.psi.PsiParenthesizedExpression;
import com.intellij.psi.PsiPrefixExpression;
import com.intellij.util.IncorrectOperationException;
import org.apache.commons.lang3.BooleanUtils;
import org.jetbrains.annotations.NotNull;

/**
 * @author zhixiao.mzx
 * @date 2021/08/02
 */
public class BooleanOperatorAction extends PsiElementBaseIntentionAction {
    public static final String ACTION_HINT_TEXT = "BooleanOperatorAction: getText";

    private PsiPrefixExpression prefixExpression;

    @Override
    public void invoke(@NotNull Project project, Editor editor, @NotNull PsiElement element)
        throws IncorrectOperationException {
        String exp = "BooleanUtils.isFalse(a)";
        PsiElementFactory factory = JavaPsiFacade.getElementFactory(project);
        PsiMethodCallExpression psiNewExpression = (PsiMethodCallExpression)factory.createExpressionFromText(exp, null);

        psiNewExpression.getArgumentList().getExpressions()[0].replace(
            Objects.requireNonNull(prefixExpression.getOperand()));

        prefixExpression.replace(psiNewExpression);
    }

    @Override
    public boolean isAvailable(@NotNull Project project, Editor editor, @NotNull PsiElement element) {
        PsiElement context = element.getContext();

        if (BooleanUtils.isFalse(context instanceof PsiPrefixExpression)) {
            context = element.getParent().getContext();
        }
        return checkPsiPrefixExpression(context);
        //
        //if (context instanceof PsiBinaryExpression && context.getChildren().length == 0) {
        //    context = context.getParent().getContext();
        //}
        //
        //if (context instanceof PsiParenthesizedExpression) {
        //    context = context.getParent();
        //}
        //
        //return checkPsiPrefixExpression(context);
    }

    /**
     * !x
     *
     * @param context
     * @return
     */
    private boolean checkPsiPrefixExpression(PsiElement context) {
        if (BooleanUtils.isFalse(context instanceof PsiPrefixExpression)) {
            return false;
        }

        prefixExpression = (PsiPrefixExpression)context;
        return Objects.equals(JavaTokenType.EXCL, prefixExpression.getOperationSign().getTokenType());
    }

    @Override
    public @NotNull @IntentionFamilyName String getFamilyName() {
        return "BooleanOperatorAction: getFamilyName";
    }

    @Override
    public @IntentionName @NotNull String getText() {
        return ACTION_HINT_TEXT;
    }
}

File: src/main/java/BooleanOperatorAction.kt
import com.intellij.codeInsight.intention.PsiElementBaseIntentionAction
import com.intellij.codeInspection.util.IntentionFamilyName
import com.intellij.codeInspection.util.IntentionName
import com.intellij.openapi.editor.Editor
import com.intellij.openapi.project.Project
import com.intellij.psi.*
import org.apache.commons.lang3.BooleanUtils

/**
 * @author zhixiao.mzx
 * @date 2021/08/02
 */
class BooleanOperatorAction : PsiElementBaseIntentionAction() {
    private var prefixExpPointer: SmartPsiElementPointer<PsiElement>? = null

    override fun invoke(project: Project, editor: Editor, element: PsiElement) {
        val prefixExpression = prefixExpPointer!!.element as PsiPrefixExpression
        var operand = prefixExpression.operand!!
        if (operand is PsiParenthesizedExpression) {
            operand = operand.expression!!
        }

        val exp = "BooleanUtils.isFalse(a)"
        val factory = JavaPsiFacade.getElementFactory(project)
        val psiNewExpression = factory.createExpressionFromText(exp, null) as PsiMethodCallExpression
        psiNewExpression.argumentList.expressions[0].replace(operand)
        prefixExpression.replace(psiNewExpression)
    }

    override fun isAvailable(project: Project, editor: Editor, element: PsiElement): Boolean {
        var context = element.context
        if (context !is PsiPrefixExpression) {
            context = element.parent.context
        }

        if (context !is PsiPrefixExpression) {
            return false
        }

        val prefixExpression = context
        if (JavaTokenType.EXCL == prefixExpression.operationSign.tokenType) {
            prefixExpPointer = SmartPointerManager.getInstance(project).createSmartPsiElementPointer(prefixExpression)
            return true
        }
        return false
    }

    override fun getFamilyName(): @IntentionFamilyName String {
        return "BooleanOperatorAction: getFamilyName"
    }

    override fun getText(): @IntentionName String {
        return ACTION_HINT_TEXT
    }

    companion object {
        const val ACTION_HINT_TEXT = "BooleanOperatorAction: getText"
    }
}